Integrations/integration-Tufin.yml

```diff
@@ -0,0 +1,321 @@
+            "service": demisto.args().get('Service', 'Any'),
+            "includeIncompletePaths": "true",
+            "displayBlockedStatus": "true"
+        }
+
+        entry = {
+            'Type': entryTypes['note'],
+            'Contents': '',
+            'ContentsFormat': formats['text'],
+            'ReadableContentsFormat': formats['markdown'],
+            'HumanReadable': ''
+        }
+
+        try:
+            img = tos_request("/securetrack/api/topology/path_image", querystring, {'accept': "image/png",'content-type': "application/json",'cache-control': "no-cache"})
+            if len(img) > 100:
+                return fileResult('topo.png', img, entryTypes['image'])
+            else:
+                entry['HumanReadable'] = 'No Valid Path Found'
+                entry['Contents'] = 'No Valid Path Found'
+                return entry
+        except:
+            entry['HumanReadable'] = 'Error Running Query'
+            entry['Contents'] = 'Error Running Query'
+            entry['Type'] = entryTypes['error']
+            return entry
+
+
+    def device_name(devices, device_id):
+        return [e['name'] + ' ({} {})'.format(e['vendor'], e['model']) for e in devices if int(e['id']) == int(device_id)][0]
+
+
+    def object_lookup():
+        """ Sample query: querystring = {"filter":"subnet","count":"50","exact_subnet":"1.1.1.1"} """
+        querystring = {
+            "filter":"subnet",
+            "count":"50",
+            "exact_subnet": demisto.args()['IP']
+        }
+
+        entry = {
+            'Type': entryTypes['note'],
+            'Contents': '',
+            'ContentsFormat': formats['json'],
+            'ReadableContentsFormat': formats['markdown'],
+            'HumanReadable': ''
+        }
+
+        return_json = {"objects": []}
+
+        o = tos_request("/securetrack/api/network_objects/search", querystring)
+
+        if o['network_objects']['count'] > 0:
+            device_json = tos_request('/securetrack/api/devices')['devices']['device']
+            objs = o['network_objects']['network_object']
+            if not isinstance(o['network_objects']['network_object'], list):
+                objs = [objs]
+            for obj in objs:
+                # display_name device_id
+                return_json['objects'].append({"object_name": obj['display_name'], "device": device_name(device_json, obj['device_id']), "comment": obj['comment']})
+        else:
+            return('Nothing Found')
+
+        # Return to Demisto
+        entry['Contents'] = json.dumps(return_json)
+        entry['HumanReadable'] = tableToMarkdown('Object Lookup for {}'.format(querystring['exact_subnet']), return_json['objects'], ['object_name', 'device', 'comment'], underscoreToCamelCase, removeNull=True)
+        return entry
+
+
+    def policy_search():
+        """ Search policy across all devices.  See docs for syntax """
+        u = '/securetrack/api/rule_search'
+        max_rules_per_device = 100
```

Comment:
Why is this hardcoded and not configurable ?

Reply:
Turned to a config parameter in next commit.